src/tests/conftest.py

```diff
@@ -1,27 +1,53 @@
+import socket
+import time
 from multiprocessing import Process
 
 import pytest
+import requests
 import uvicorn
 
 from fastapi_app import seed_data
 from fastapi_app.app import app
 
 
-def run_server():
-    uvicorn.run(app)
+def wait_for_server_ready(url: str, timeout: float = 10.0, check_interval: float = 0.5) -> bool:
+    """Make requests to provided url until it responds without error."""
+    conn_error = None
+    for _ in range(int(timeout / check_interval)):
+        try:
+            requests.get(url)
+        except requests.ConnectionError as exc:
+            time.sleep(check_interval)
+            conn_error = str(exc)
+        else:
+            return True
+    raise RuntimeError(conn_error)
 
 
 @pytest.fixture(scope="session")
-def live_server():
-    seed_data.load_from_json()
-    proc = Process(target=run_server, daemon=True)
-    proc.start()
-    yield
-    proc.kill()
-    seed_data.drop_all()
+def free_port() -> int:
+    """
+    Return a free port on localhost
+    """
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        addr = s.getsockname()
```

Comment:
Looks like I do it slightly differently in https://github.com/Azure-Samples/azure-search-openai-demo/blob/18eee6bc1251c6eef1366a66df310865b027fc34/tests/e2e.py#L32 maybe setsockopt is needed for CI?